### PR TITLE
Fix checkstyle violations in `Console`

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 
Fix checkstyle violations in `Console`. Refactored to move functionality out in to more targeted classes.

### Testing done 
_Describe the testing stategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java [hunk 1]
```diff
@@ -110,7 +166,7 @@ public Console(final OutputFormat outputFormat, final KsqlRestClient restClient)
 
   protected abstract LineReader buildLineReader();
 
-  protected abstract void puts(InfoCmp.Capability capability);
+  public abstract void puts(InfoCmp.Capability capability);
```
Comment (dguy) [inline ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java]: This is only used in one place AFAICT. Maybe we should just make it `clear()` and then not expose the `JLine` specific enum?

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java [hunk 2]
```diff
@@ -377,117 +286,15 @@ private void printAsTable(final GenericRow row) {
...
-      tableBuilder.withColumnHeaders("Function Name", "Type");
-      ((FunctionNameList) ksqlEntity)
-          .getFunctions().stream().sorted()
-          .forEach(func -> tableBuilder.withRow(
-              Arrays.asList(func.getName(), func.getType().name().toUpperCase())));
-    } else if (ksqlEntity instanceof FunctionDescriptionList) {
-      printFunctionDescription((FunctionDescriptionList) ksqlEntity);
-      return;
-    } else {
+    final Handler<KsqlEntity, Console> defaultHandler = (console, entity) -> {
```
Comment (dguy) [inline ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java]: Do we actually need to have a defaultHandler? i.e, at the moment there is only 1 and the `getOrDefault` method could just be `get` and throw the exception? Doesn't seem like it is really necessary at the moment

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/Clear.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  @Override
+  public void printHelp() {
+    console.writer().println("clear:");
+    console.writer().println("\tClear the current terminal.");
+  }
+
+  @Override
+  public void execute(final String commandStrippedLine) {
+    console.puts(InfoCmp.Capability.clear_screen);
```
Comment (dguy) [inline ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/Clear.java]: See comment above.

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/Exit.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  public void printHelp() {
+    console.writer().println("exit:");
+    console.writer().println(
+        "\tExit the CLI."
+    );
+  }
+
+  @Override
+  public void execute(final String commandStrippedLine) {
+    throw new EndOfFileException();
```
Comment (dguy) [inline ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/Exit.java]: This seems odd. Using an exception for non exceptional behaviour

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/Help.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  @Override
+  public String getName() {
+    return "help";
+  }
+
+  @Override
+  public void printHelp() {
+    console.writer().println("help:");
```
Comment (dguy) [inline ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/Help.java]: Not really part of this PR, but i wish we didn't have to do `console.writer().println(...)`...

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/History.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        "history:");
+    console.writer().println(
+        "\tShow previous lines entered during the current CLI session. You can"
+            + " use up and down arrow keys to view previous lines."
+    );
+  }
+
+  @Override
+  public void execute(final String commandStrippedLine) {
+    for (final org.jline.reader.History.Entry historyEntry : console.getLineReader().getHistory()) {
```
Comment (dguy) [inline ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/History.java]: kinda seems like there could be a `console.printHistory()` method

File: ksql-common/src/main/java/io/confluent/ksql/util/HandlerMap.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+import com.google.common.collect.Maps;
+import java.util.Map;
+
+/**
+ * Type-safe collection of handlers for different types.
+ *
+ * @param <T> the type to be handled.
+ * @param <A> the type of the argument that is passed to the handler along with the type.
+ */
+public final class HandlerMap<T, A> {
```
Comment (dguy) [inline ksql-common/src/main/java/io/confluent/ksql/util/HandlerMap.java]: Tests?

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java [hunk 8]
```diff
@@ -74,12 +81,60 @@
...
+          .put(ExecutionPlan.class,
+              tablePrinter(ExecutionPlanTableBuilder.class, ExecutionPlan.class))
+          .put(FunctionNameList.class,
+              tablePrinter(FunctionNameListTableBuilder.class, FunctionNameList.class))
+          .put(FunctionDescriptionList.class,
+              Console::printFunctionDescription)
+          .build();
+
+  private static <T extends KsqlEntity> Handler<KsqlEntity, Console> tablePrinter(
+      final Class<? extends TableBuilder<T>> tableBuilderType,
```
[the remark this answers is not in the document]
Comment (big-andy-coates) [reply ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java]: As in `Sometype::new` - good idea!

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/Exit.java (hunk 4 above)
Comment (big-andy-coates) [reply ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/Exit.java]: Not my code - just a refactor.

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java (hunk 1 above)
Comment (big-andy-coates) [reply ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java]: Makes sense. (Was trying to minimise changes....)

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java (hunk 2 above)
Comment (big-andy-coates) [reply ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java]: Yeah, I'd spotted the same.

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/Clear.java (hunk 3 above)
Comment (big-andy-coates) [reply ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/Clear.java]: done.

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/Help.java (hunk 5 above)
Comment (big-andy-coates) [reply ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/Help.java]: Agreed

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/History.java (hunk 6 above)
Comment (big-andy-coates) [reply ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/History.java]: Done.

File: ksql-common/src/main/java/io/confluent/ksql/util/HandlerMap.java (hunk 7 above)
Comment (big-andy-coates) [reply ksql-common/src/main/java/io/confluent/ksql/util/HandlerMap.java]: Your wish...

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java [hunk 9]
```diff
@@ -40,48 +54,87 @@
...
+          .build();
+
+  private static <T extends KsqlEntity> Handler<KsqlEntity, Console> tablePrinter(
+      final Class<T> entityType,
+      final Supplier<? extends TableBuilder<T>> tableBuilderType) {
+
+    try {
+      final TableBuilder<T> tableBuilder = tableBuilderType.get();
+
+      return (console, type) -> {
```
Comment (apurvam) [inline ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java]: nit: should `type` be called `ksqlEntity` instead?

Comment (big-andy-coates) [reply ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java]: That's not a nit, it's a uber nit!
